src/parser.rs

```diff
@@ -1110,6 +1111,12 @@ impl Parser {
         let (columns, constraints) = self.parse_columns()?;
         let with_options = self.parse_with_options()?;
 
+        let query = if self.parse_keyword(Keyword::AS) {
+            Some(Box::new(self.parse_query()?))
```

Comment:
You mean options? Not sure I follow. I believe `AS query` always comes last, e.g see in bigquery (supposed to be ansi sql) where it comes after the options:

https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_table_statement

Reply:
Also create external table will not have `AS` in any case I guess?

Reply:
To better convince myself: `WITH ( options )` is from #74, which cited PostgreSQL and MSSQL as supporting it. From your link I see that BigQuery has `OPTIONS(...)` which is similar, but which we don't support. Both come before `AS`, like you said.

([ANSI SQL](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#_11_3_table_definition) doesn't have that. While it has `WITH [NO] DATA` and `WITH SYSTEM VERSIONING`, both after the `AS <query>`, we don't support neither.)

Reply:
The ColumnOptionDef was an example of an alternative approach, where instead of convincing ourselves all the major dialects agree on the ordering, we implement something more generic.

You're right that at this moment `EXTERNAL` and `AS <query>` are mutually exclusive in Hive, but `STORED AS <file_format>` and other options apply to both. As you can see, the PR #46 originally had two separate variants. They were combined into one during review, but the two parser methods were kept separate, though it's clearly inconsistent.

Reply:
Ok! So at least for the purpose of this PR, it needs to go before: with options.

For the ordering, I think it indeed makes sense to make it less restrictive, but maybe would better be handled in another PR?

Reply:
Yes on not refactoring in this PR.

Just to be clear (as I'm not sure I understood your latest comment correctly), I believe the current order to be correct, though perhaps we should add a comment clarifying why: of the dialects we considered (PostgreSQL, MSSQL, BigQuery), only PostgreSQL supports `WITH ( options )` with `AS` and documents `WITH (...)` to come before the `AS`.

Reply:
Ah, misread earlier comment and ANSI SQL doc. Will add some comments here as well.